kernel/contracts/ZepCore.sol

```diff
@@ -3,77 +3,60 @@ pragma solidity ^0.4.18;
+    
     developerFraction = _developerFraction;
     newVersionCost = _newVersionCost;
     // TODO: we need to think how we are going to manage variable costs to propose new versions
   }
 
   function register(KernelInstance instance) public {
-    registry().addInstance(instance);
-    token().transferFrom(msg.sender, this, newVersionCost);
-    token().burn(newVersionCost);
-    // TODO: should we build a burnFrom or sth like that?
+    _registry.addInstance(instance);
+    
+    // TODO: Update to burnFrom once https://github.com/OpenZeppelin/zeppelin-solidity/pull/870 is merged
+    _token.transferFrom(msg.sender, this, newVersionCost);
+    _token.burn(newVersionCost);
   }
 
-  function transferStake(KernelInstance from, KernelInstance to, uint256 amount, bytes data) public {
-    // TODO: an unstake followed by a stake call won't work
-    require(false);
+  function getInstance(string name, string version) public view returns(KernelInstance) {
+    return _registry.getInstance(name, version);
   }
 
   function stake(KernelInstance instance, uint256 amount, bytes data) public {
-    token().transferFrom(msg.sender, this, amount);
+    _token.transferFrom(msg.sender, this, amount);
     uint256 developerPayout = amount.div(developerFraction);
     require(developerPayout > 0);
-    // TODO: check how we can manage remainders in a better way
+    // TODO: Think how we can manage remainders in a better way
 
     uint256 stakedAmount = amount.sub(developerPayout);
-    _totalStaked = totalStaked().add(stakedAmount);
-    _instanceVouches[instance] = totalStakedFor(instance).add(stakedAmount);
-    _userVouches[msg.sender][instance] = _userVouches[msg.sender][instance].add(stakedAmount);
-    token().transfer(instance.developer(), developerPayout);
-
-    Staked(msg.sender, instance, stakedAmount, totalStakedFor(instance), data);
+    _stakes.stake(msg.sender, instance, stakedAmount, data);
+    _token.transfer(instance.developer(), developerPayout);
   }
 
   function unstake(KernelInstance instance, uint256 amount, bytes data) public {
-    uint256 currentStake = _userVouches[msg.sender][instance];
-    require(currentStake >= amount);
-
-    _totalStaked = totalStaked().sub(amount);
-    _instanceVouches[instance] = totalStakedFor(instance).sub(amount);
-    _userVouches[msg.sender][instance] = currentStake.sub(amount);
-    token().transfer(msg.sender, amount);
-    
-    Unstaked(msg.sender, instance, amount, totalStakedFor(instance), data);
+    _stakes.unstake(msg.sender, instance, amount, data);
+    _token.transfer(msg.sender, amount);    
   }
 
-  function totalStakedFor(KernelInstance instance) public view returns (uint256) {
-    return _instanceVouches[instance];
-  }
-
-  function totalStaked() public view returns (uint256) {
-    return _totalStaked;
+  function transferStake(KernelInstance from, KernelInstance to, uint256 amount, bytes data) public {
+    _stakes.transferStake(msg.sender, from, to, amount, data);
```

Comment:
Oh! But the developer of the `to` KernelInstance is not getting her share of tokens, is she?

Reply:
This is why code reviews are needed. Thanks!